src/heitech.zer0mqXt.core/Patterns/Retry.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Threading.Tasks;
+using heitech.zer0mqXt.core.infrastructure;
+
+namespace heitech.zer0mqXt.core.patterns
+{
+    ///<summary>
+    /// Encapsulates the retry policy grasped from the configuration
+    ///</summary>
+    internal class Retry
+    {
+        private readonly SocketConfiguration _configuration;
+
+        internal Retry(SocketConfiguration configuration)
+        {
+            this._configuration = configuration;
+        }
+
+        ///<summary>
+        /// Register your retryable callback here
+        ///</summary>
+        internal async Task<XtResult<T>> RunAsyncWithRetry<T>(Func<Task<XtResult<T>>> retryableAction, string operationName = "request")
+            where T : class
+        {
+            try
+            {
+                return await retryableAction().ConfigureAwait(false);
+            }
+            catch (NetMQ.EndpointNotFoundException ntfnd)
+            {
+                _configuration.Logger.Log(new ErrorLogMsg($"NetMQ.Endpoint could not be found at {_configuration.Address()}: " + ntfnd.Message));
+                await Task.Delay((int)_configuration.TimeOut.TotalMilliseconds).ConfigureAwait(false);
+                try
+                {
+                    return await retryableAction().ConfigureAwait(false);
```

Comment:
Should we also introduce a `RetryCount` which may default to one? Perhaps in a follow-up issue.
Or is a single retry absolutely covering the typical cases?

Reply:
Yeah sounds good to me, maybe you´d like to add an issue for that?

Reply:
Yes I would: see #52

Reply:
ty :)